python/desc/skycatalogs/skyCatalogs.py

```diff
@@ -99,15 +99,30 @@ class SkyCatalog(object):
     point sources, SSOs
 
     '''
-    def __init__(self, config, mp=False, verbose=False):
+    def __init__(self, config, mp=False, skycatalog_root=None, verbose=False):
         '''
         Parameters
         ----------
         config:  dict.  Typically the result of loading a yaml file
         mp:      boolean  Default False. Set True to enable multiprocessing.
+        skycatalog_root: If not None, overrides value in config or
+                         in environment variable SKYCATALOG_ROOT
         '''
         self._config = config
         self._mp = mp
+        if 'schema_version' not in config:
+            self._cat_dir = config['root_directory']
+        else:
+            sky_root = config['skycatalog_root']        # default
```

Comment:
This all seems fine to me, though I would consider setting this default value only after the keyword option and environment variable have been checked so that `skycatalog_root` could actually be omitted from the config yaml file.   It would be useful to use python logging and have a logger print the eventual value of `self._cat_dir` at the `logging.INFO` or `logging.DEBUG` level.

Reply:
I'll see about adding the logging code to print out the ultimate resolution for self._cat_dir.  That could be useful if things go awry.
 I was planning to have the sky catalog creator write out `skycatalog_root` and `catalog_dir` as specified by command-line options, so that `skycatalog_root` would always be there.  During creation it could default to something (e.g., current directory), but my intention was to still write out the corresponding absolute path, so there would be a record in the config of where the data were written originally.   Of course, you would have to know where the config was written; I can't say I've entirely thought this through.  Maybe I should also log one or both of config file location and data location when catalogs are created.